src/database/test/DatabaseTests.cpp

```diff
@@ -377,13 +377,56 @@ class SchemaUpgradeTestApplication : public TestApplication
 
 TEST_CASE("schema upgrade test", "[db]")
 {
-    auto test_one_db_mode = [](Config::TestDbMode const db_mode) {
+    auto prepOldSchemaDB = [](SchemaUpgradeTestApplication& app) {
+        auto& session = app.getDatabase().getSession();
+        {
+            std::string const mAccountID = "account";
+            int64_t const mBalance = 500;
+            int64_t const mSeqNum = 7;
+            int32_t const mSubEntryNum = 0;
+            std::string const mInflationDest = "infdest";
+            std::string const mHomeDomain = "my.home";
+            std::string const mThreshold = "threshold";
+            std::string const mSigners = "autograph";
+            int32_t const mFlags = 0;
+            int32_t const mLastModified = 3;
+            int64_t const mBuyingLiabilities = 12;
+            int64_t const mSellingLiabilities = 17;
+
+            soci::transaction tx(session);
+
+            // Use raw SQL to perform a couple of database operations,
+            // since we're writing in an old database format, and calling
+            // standard interfaces to create accounts or trustlines would
+            // use SQL corresponding to the new format to which we'll
+            // soon upgrade.
+            session
+                << "INSERT INTO accounts ( "
+                   "accountid, balance, seqnum, numsubentries, inflationdest,"
+                   "homedomain, thresholds, signers, flags, lastmodified, "
+                   "buyingliabilities, sellingliabilities "
+                   ") VALUES ( "
+                   ":id, :v1, :v2, :v3, :v4, :v5, :v6, :v7, :v8, :v9, :v10, "
+                   ":v11 "
+                   ")",
+                soci::use(mAccountID, "id"), soci::use(mBalance, "v1"),
```

Comment:
You should use positional substitution instead of named-value substitution, it's considerably faster. Obviously this is a test so performance is much less important, but that is the reason we use positional substitution everywhere else.

Reply:
Ah, I somehow failed to notice that I was doing that and the existing code wasn't.  Fixed.